src/Components/Join/Join.jsx

```diff
@@ -25,10 +27,61 @@ class Join extends React.Component {
 
     componentDidMount() {
         document.title = 'Deeper Blue: Join Game';
+        this.getGames();
+        this.activeGameUsernames();
         this.httpGetRequest('/gamelist');
         this.initAuthListener();
     }
 
+    getGames() {
```

Comment:
I'm not completely sure how this page works - you'll have to ask Daniel, but it seems like `this.httpGetRequest('/gamelist')` and `parse` already gets the games from the server and populates the `game_list` in the state object. 

The server picks out games based on some criteria:

```python
# Snippet of `/gamelist` query from the server
@app.route('/gamelist')
def game_list():
    query = db.collection(GAMES_COLLECTION) \
        .where('game_over.game_over', '==', False) \
        .where('free_slots', '>', 0) \
        .where('public', '==', True)
return jsonify([ref.to_dict() for ref in query.get()])
```

So maybe you can use `game_list` instead of doing another Firebase query again with `getGames()`? 

This would also possibly eliminate the error screenshots I posted on the PR, because they seem to be related to how Firebase returns query snapshots or something like that.

Reply:
The problem with game_list is that it only returns the games with at least 1 open slot. For the "Active games" table we want all games where the user is one of the players regardless if there are any open slots left.

Reply:
Ahh I see